src/node_url.cc

```diff
@@ -1545,44 +1546,61 @@ void URL::Parse(const char* input,
 }  // NOLINT(readability/fn_size)
 
 // https://url.spec.whatwg.org/#url-serializing
-std::string URL::SerializeURL(const struct url_data* url,
+std::string URL::SerializeURL(const url_data& url,
                               bool exclude = false) {
-  std::string output = url->scheme;
-  if (url->flags & URL_FLAGS_HAS_HOST) {
+  std::string output;
+  output.reserve(
+    10 +
+    url.scheme.size() +
+    url.username.size() +
+    url.password.size() +
+    url.host.size() +
+    url.query.size() +
+    url.fragment.size() +
+    url.href.size() +
+    std::accumulate(
+        url.path.begin(),
+        url.path.end(),
+        0,
+        [](size_t sum, const auto& str) { return sum + str.size(); }));
```

Comment:
nit:
```suggestion
        [](size_t sum, std::string_view str) { return sum + str.size(); }));
```
or did you choose `const auto&` because it's shorter?

Reply:
See the "Using `auto`" section in the C++ style guide

Reply:
@benjamingr I saw it. `const auto&` translates to `const std::string&` in this case, which is not the same as a `std::string_view`. To use a `std::string_view` here, we need to explicitly set the type instead of relying on auto.

I suggested using `std::string_view` here because we are gradually moving things to C++17, like in https://github.com/nodejs/node/pull/41757 (this adds another `std::string_view`).

Reply:
The change is quite inconsequential here, while in #41757 changing it to `string_view` allows more flexibility in callers.

Reply:
@RaisinTen Since this is an actual bugfix, I’d like to keep this clear of C++17 changes. Also, as @TimothyGu says, it’s very inconsequential here.

Reply:
Makes sense to me :+1: